test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,146 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+			}
+
+			ipPool, err := crdClient.CrdV1beta1().IPPools().Get(ctx, ipPoolName[0], metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkIPsAndIfaces(before, after map[string]net.IP) bool {
+	if len(before) != len(after) {
+		return false
+	}
+	for iface, beforeIP := range before {
+		afterIP, exists := after[iface]
+		if !exists || !beforeIP.Equal(afterIP) {
+			return false
+		}
+	}
+	return true
+}
+
+// reconcilationAfterAgentRestart verifies Interface cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	for _, pod := range data.pods {
+		beforeAgentRestartIPsAndIfaces, err := data.listPodIPs(pod)
+		if err != nil {
+			return fmt.Errorf("failed to get Pod IP before agent restart: %w", err)
+		}
+
+		// Restarting the Antrea agent
+		if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+			t.Fatalf("Failed to restart Antrea agent Pods: %v", err)
+		}
+
+		afterAgentRestartIPsAndIfaces, err := data.listPodIPs(pod)
+		if err != nil {
+			return fmt.Errorf("failed to get Pod IP after agent restart: %w", err)
+		}
+
+		if !checkIPsAndIfaces(beforeAgentRestartIPsAndIfaces, afterAgentRestartIPsAndIfaces) {
+			return fmt.Errorf("Pod: %s, Interfaces and IPs mismatch after agent restart", pod.podName)
+		}
+
+		logs.Infof("Pod %v IPs before agent restart:", pod.podName)
+		for iface, ip := range beforeAgentRestartIPsAndIfaces {
+			logs.Infof("  Interface: %s, IP: %s", iface, ip)
+		}
+
+		logs.Infof("Pod %v IPs after agent restart:", pod.podName)
+		for iface, ip := range afterAgentRestartIPsAndIfaces {
+			logs.Infof("  Interface: %s, IP: %s", iface, ip)
+		}
+	}
+	return data.deleteVLANPod(t)
+}
+
+func (data *testData) deleteVLANPod(t *testing.T) error {
+	vlanPod := data.pods[1]
+	interfaces := []string{"eth1", "eth2"}
+	beforePodDeletionIPsAndIfaces, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get Pod IP: %w", err)
+	}
+
+	ipPools, err := data.getIPPoolNames(interfaces, vlanPod, "default")
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool: %w", err)
+	}
+
+	// Remove Pod and check IP released or not.
+	if err := data.e2eTestData.DeletePodAndWait(defaultTimeout, vlanPod.podName, data.e2eTestData.GetTestNamespace()); err != nil {
+		return fmt.Errorf("failed to delete Pod %s: %w", vlanPod.podName, err)
+	}
+	afterPodDeletionIPsAndIfacess, err := data.listPodIPs(vlanPod)
+	if len(afterPodDeletionIPsAndIfacess) != 0 || err == nil {
+		return fmt.Errorf("Pod interfaces and IP still exist")
+	}
+
+	return data.checkIPReleased(ipPools, beforePodDeletionIPsAndIfaces, interfaces)
```

Comment:
Still I did not see you check OVS ports are deleted?

Reply:
my bad,  done please check.